develop does not compile with Boost 1.72

Presumably broken by #2636.

```
[ 18%] Building CXX object src/IO/CMakeFiles/IO.dir/H5/Helpers.cpp.o
In file included from /home/wthrowe/spectre/src/IO/H5/Helpers.cpp:13:
In file included from /home/wthrowe/spectre/src/DataStructures/BoostMultiArray.hpp:20:
In file included from /usr/include/boost/multi_array.hpp:29:
In file included from /usr/include/boost/multi_array/base.hpp:22:
In file included from /usr/include/boost/multi_array/extent_gen.hpp:18:
/usr/include/boost/multi_array/types.hpp:29:24: error: type alias redefinition with different types ('std::ptrdiff_t' (aka 'long') vs 'std::size_t' (aka 'unsigned long'))
typedef std::ptrdiff_t index;
                       ^
/home/wthrowe/spectre/src/DataStructures/BoostMultiArray.hpp:15:7: note: previous definition is here
using index = std::size_t;
      ^
1 error generated.
```

Examining the compiler invocation, the flag `-DBOOST_MULTI_ARRAY_TYPES_HPP` is being passed, which is the value for Boost 1.73.

From CMake output:
```
-- Found Boost: /usr/include (found suitable version "1.72.0", minimum required is "1.60.0") found components: program_options 
-- Boost include: /usr/include
-- Boost libraries: /usr/lib64/libboost_program_options-mt.so
```

LibraryVersions.txt
```
CMake Version: 3.17.4
SpECTRE Version: 2020.12.7
SpECTRE branch: develop
SpECTRE hash: v2020.12.07
Hostname: ungoliant
Host system: Linux-5.4.60-gentoo
Host system version: 5.4.60-gentoo
Host system processor: x86_64
CMAKE_CXX_FLAGS: -fPIC -fdiagnostics-color=always -ferror-limit=1 -ftemplate-backtrace-limit=0 -march=native
CMAKE_CXX_LINK_FLAGS:  -L/usr/lib64 -fuse-ld=gold
CMAKE_CXX_FLAGS_DEBUG: -g -DSPECTRE_DEBUG
CMAKE_CXX_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_C_FLAGS: -fPIC -march=native
CMAKE_C_FLAGS_DEBUG: -g
CMAKE_C_FLAGS_RELEASE: -O3 -DNDEBUG
CMAKE_Fortran_FLAGS: -march=native
CMAKE_Fortran_FLAGS_DEBUG: -g
CMAKE_Fortran_FLAGS_RELEASE: -O3 -DNDEBUG -O3
CMAKE_EXE_LINKER_FLAGS: -march=native -no-pie
CMAKE_BUILD_TYPE: Debug
CMAKE_CXX_COMPILER: /usr/lib/llvm/10/bin/clang++-10
CMAKE_CXX_COMPILER_VERSION: 10.0.1
CMAKE_C_COMPILER: /usr/lib/llvm/10/bin/clang-10
CMAKE_C_COMPILER_VERSION: 10.0.1
CMAKE_Fortran_COMPILER: /usr/bin/gfortran
CMAKE_Fortran_COMPILER_VERSION: 7.5.0
Python Version: 3.7.9
```

--- a/cmake/SetupBoost.cmake
+++ b/cmake/SetupBoost.cmake
@@ -3,12 +3,15 @@
 
 find_package(Boost 1.60.0 REQUIRED COMPONENTS program_options)
 
+# CMake versions don't set this consistently
+set(Boost_VERSION "${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION}.${Boost_SUBMINOR_VERSION}")
+
 message(STATUS "Boost include: ${Boost_INCLUDE_DIRS}")
 message(STATUS "Boost libraries: ${Boost_LIBRARIES}")
 
 file(APPEND
   "${CMAKE_BINARY_DIR}/LibraryVersions.txt"
-  "Boost Version:  ${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION}.${Boost_SUBMINOR_VERSION}\n"
+  "Boost Version:  ${Boost_VERSION}\n"
   )
 
 # Boost organizes targets as:

